rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -321,6 +350,35 @@ private void processBlockMessage(@Nonnull final MessageChannel sender, @Nonnull
         tryRelayBlock(sender, block, result);
         recordEvent(sender, EventType.VALID_BLOCK);
         Metrics.processBlockMessage("finish", block, sender.getPeerNodeID());
+
+        // If still synchronizing then return
+        if (blockProcessor.hasBetterBlockToSync()) {
+            return;
+        }
+
+        // If we are a valid source for federation notifications then generate one
+        if (federationNotificationSource != null) {
+            federationNotificationSource.generateNotification();
+        }
+
+        // Check if the Federation disappeared for a long time (Federation eclipsed).
+        federationNotificationProcessor.checkIfFederationWasEclipsed();
```

Comment:
I would delegate this call to a more specific component, unless there's a good reason to do it here.